test/extended/authorization/podsecurity_admission.go

```diff
@@ -25,16 +40,185 @@ var _ = g.Describe("[sig-auth][Feature:PodSecurity]", func() {
 			},
 			Spec: corev1.PodSpec{
 				Containers: []corev1.Container{
-					{
-						Name:    "sleeper",
-						Image:   "fedora:latest",
-						Command: []string{"sleep"},
-						Args:    []string{"infinity"},
-					},
+					sleeperContainer,
 				},
 			},
 		}, metav1.CreateOptions{})
 		o.Expect(err).NotTo(o.HaveOccurred())                                                     // the pod passed admission
 		o.Expect(pod.Annotations[securityv1.ValidatedSCCAnnotation]).To(o.Equal("restricted-v2")) // and the mutating SCC is restricted-v2
 	})
 })
+
+var _ = g.Describe("[sig-auth][Feature:PodSecurity][Feature:SCC]", func() {
+	defer g.GinkgoRecover()
+
+	oc := exutil.NewCLIWithPodSecurityLevel("pod-security-scc-mutation", psapi.LevelRestricted)
+
+	g.It("creating pod controllers", func() {
+		ns, err := oc.AdminKubeClient().CoreV1().Namespaces().Get(context.Background(), oc.Namespace(), metav1.GetOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		ns.Labels[psapi.EnforceLevelLabel] = "privileged" // we're interested only in warnings here
+		_, err = oc.AdminKubeClient().CoreV1().Namespaces().Update(context.Background(), ns, metav1.UpdateOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		userConfig := rest.CopyConfig(oc.UserConfig())
+		recorder := &HeaderRecordingRoundTripper{}
+		userConfig.Wrap(recorder.Wrap)
+
+		kubeClient, err := kubernetes.NewForConfig(userConfig)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		privilegedSA, err := oc.AdminKubeClient().CoreV1().ServiceAccounts(oc.Namespace()).Create(
+			context.Background(),
+			&corev1.ServiceAccount{
+				ObjectMeta: metav1.ObjectMeta{
+					GenerateName: "privileted-sa-",
+				},
+			},
+			metav1.CreateOptions{},
+		)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		_, err = oc.AdminKubeClient().RbacV1().RoleBindings(oc.Namespace()).Create(context.Background(),
+			&v1.RoleBinding{
+				ObjectMeta: metav1.ObjectMeta{
+					GenerateName: "privileged-scc-",
+				},
+				RoleRef: v1.RoleRef{
+					APIGroup: "rbac.authorization.k8s.io",
+					Kind:     "ClusterRole",
+					Name:     "system:openshift:scc:privileged",
+				},
+				Subjects: []v1.Subject{
+					{
+						Kind:      "ServiceAccount",
+						Name:      privilegedSA.Name,
+						Namespace: oc.Namespace(),
+					},
+				},
+			},
+			metav1.CreateOptions{},
+		)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		for _, tt := range []struct {
+			name           string
+			container      corev1.Container
+			saName         string
+			expectWarnings bool
+		}{
+			{
+				name:           "SHOULD throw warnings if no SCC matches the pod template",
+				container:      privilegedContainer(sleeperContainer),
+				saName:         "default",
+				expectWarnings: false, // the PodSpecExtractor currently fails in this case and so nothing is thrown/logged in audits
```

Comment:
@deads2k this is due to https://github.com/openshift/kubernetes/commit/6fe5c8f28c24dc9de63cf54138a7a1adc3bc3b78#diff-7db8f52771920b1336fa638b8e08703795c29e33d6c242f7246f707ca3cb3048R99

Do we want to ignore workloads that don't match an SCC for audit/warnings?

Reply:
So no SCC matched, we returned an error, the pod would not validate, and no warning appeared?

Reply:
No SCC matched, we returned an error.

The admission exits at that point (https://github.com/kubernetes/kubernetes/blob/03b78f3a15a69d72efd546ed867a46799f196a4b/staging/src/k8s.io/pod-security-admission/admission/admission.go#L435-L444), possibly waiting for the actual pod to appear.

No warnings are returned.